UnitTestApplicationTests/LoginViewControllerTests.swift

```diff
@@ -35,106 +35,40 @@ class LoginViewControllerTests: XCTestCase {
-        viewController.successLabel.text = nil
-        viewController.signInButton.isHidden = true
-        viewController.spinner.startAnimating()
-        //When
-        viewController.userProvider?.failure?(nil)
-        //Then
-        XCTAssertEqual(viewController.successLabel.text, "Couldn't login 😱", "Should contain \"Couldn't login 😱\"")
-        XCTAssertEqual(viewController.signInButton.isHidden, false)
-        XCTAssertEqual(viewController.spinner.isAnimating, false)
-    }
     
-    func testThatLoginTextFieldValidatorsAreSetUpOnViewDidLoad() {
-        viewController.viewDidLoad()
-        
-        XCTAssertEqual(viewController.loginTextField.validators?.count, 2, "Login text field should have 2 validators assigned")
-        XCTAssert(viewController.loginTextField.validators?.first is IsEmptyValidator, "Login text field should have Is Empty Validator assigned")
-        XCTAssert(viewController.loginTextField.validators?.last is IsEmailValidator, "Login text field should have Is Email Validator assigned")
-    }
-
-    func testThatLoginIsEmptyValidatorIsValidActionIsSetToHideValidationLabel() {
-        viewController.viewDidLoad()
-        viewController.loginValidationLabelHeightConstraint.constant = 999
-        
-        viewController.loginTextField.validators?.first?.isValidAction?()
-        
-        XCTAssertEqual(viewController.loginValidationLabelHeightConstraint.constant, 0, "Is empty validator should hide validation label on success")
-    }
-
-    func testThatLoginIsEmptyValidatorIsInvalidActionIsSetToShowValidationLabelWithValidationText() {
+    func testThatLoginIsEmptyValidatorIsSetOnViewDidLoad() {
+        //Given
         viewController.viewDidLoad()
         viewController.loginValidationLabel.text = nil
         viewController.loginValidationLabelHeightConstraint.constant = 0
-        
-        viewController.loginTextField.validators?.first?.isInvalidAction?()
-        
+        //When
+        viewController.signIn()
+        //Then
         XCTAssertEqual(viewController.loginValidationLabelHeightConstraint.constant, viewController.labelDefaultHeight, "Is empty validator should show validation label on failure")
         XCTAssertEqual(viewController.loginValidationLabel.text, "Login cannot be empty", "Is empty validator should show validation text on failure")
     }
-
-    func testThatLoginIsEmailValidatorIsValidActionIsSetToHideValidationLabel() {
-        viewController.viewDidLoad()
-        viewController.loginValidationLabelHeightConstraint.constant = 999
-        
-        viewController.loginTextField.validators?.last?.isValidAction?()
-        
-        XCTAssertEqual(viewController.loginValidationLabelHeightConstraint.constant, 0, "Is empty validator should hide validation label on success")
-    }
     
-    func testThatLoginIsEmailValidatorIsInvalidActionIsSetToShowValidationLabelWithValidationText() {
+    func testThatLoginIsEmailValidatorIsSetOnViewDidLoad() {
```

Comment:
I was thinking about this and in this case I actually want to make sure that all validators are set and present - and the way I'm checking it is by calling sign in (thanks to your advice). This assertions are saying - this is set. If this is not enormous flaw in your opinion I would leave this as is.